lua/lspconfig/configs.lua

```diff
@@ -40,6 +40,7 @@ function configs.__newindex(t, config_name, config_def)
       commands = { config.commands, 't', true },
     }
     if config.commands then
+      vim.deprecate('lspconfig.[server].setup.commands', 'vim.api.nvim_create_user_command', '0.9', 'nvim-lspconfig')
```

Comment:
While I think this deprecation message makes a lot of sense to have, it's pretty disruptive at the moment. Especially considering that some of the provided server configurations still make use of this API (i.e. nothing a user can fix themselves). May I suggest breaking the deprecation down in steps?

1) Soft-soft deprecation (deprecation notices in documentation and release notes only). This will allow 3rd party plugins time to move away from it (rust-tools.nvim is a pretty popular one, as an example).
2) Soft deprecation via warning message.
3) Removal of the feature.

Reply:
Removed the notification. I have added something in the commit message + in `lspconfig.txt`

Also, I have been running this branch from when it was started by mjlbach. Have had no issues. 
In fact, it improves a lot of usage options as `nvim_exec_autocmd` for `lspconfig` can now be called if you are defining/configuring a server not on `FileType` event.